tests/components/bluesound/conftest.py

```diff
@@ -1,71 +1,122 @@
 """Common fixtures for the Bluesound tests."""
 
-from collections.abc import Generator
+from collections.abc import AsyncGenerator, Generator
+from dataclasses import dataclass
+import ipaddress
+from typing import Any
 from unittest.mock import AsyncMock, patch
 
-from pyblu import Status, SyncStatus
+from pyblu import Input, Player, Preset, Status, SyncStatus
 import pytest
 
 from homeassistant.components.bluesound.const import DOMAIN
 from homeassistant.const import CONF_HOST, CONF_PORT
 from homeassistant.core import HomeAssistant
 
+from .utils import ValueStore
+
 from tests.common import MockConfigEntry
 
 
-@pytest.fixture
-def sync_status() -> SyncStatus:
-    """Return a sync status object."""
-    return SyncStatus(
-        etag="etag",
-        id="1.1.1.1:11000",
-        mac="00:11:22:33:44:55",
-        name="player-name",
-        image="invalid_url",
-        initialized=True,
-        brand="brand",
-        model="model",
-        model_name="model-name",
-        volume_db=0.5,
-        volume=50,
-        group=None,
-        master=None,
-        slaves=None,
-        zone=None,
-        zone_master=None,
-        zone_slave=None,
-        mute_volume_db=None,
-        mute_volume=None,
-    )
+@dataclass
+class PlayerMockData:
+    """Container for player mock data."""
 
+    host: str
+    player: AsyncMock
+    status_store: ValueStore[Status]
+    sync_status_store: ValueStore[SyncStatus]
 
-@pytest.fixture
-def status() -> Status:
-    """Return a status object."""
-    return Status(
-        etag="etag",
-        input_id=None,
-        service=None,
-        state="playing",
-        shuffle=False,
-        album=None,
-        artist=None,
-        name=None,
-        image=None,
-        volume=10,
-        volume_db=22.3,
-        mute=False,
-        mute_volume=None,
-        mute_volume_db=None,
-        seconds=2,
-        total_seconds=123.1,
-        can_seek=False,
-        sleep=0,
-        group_name=None,
-        group_volume=None,
-        indexing=False,
-        stream_url=None,
-    )
+    @staticmethod
+    async def generate(host: str) -> "PlayerMockData":
+        """Generate player mock data."""
+        host_ip = ipaddress.ip_address(host)
+        assert host_ip.version == 4
+        mac_parts = [0xFF, 0xFF, *host_ip.packed]
+        mac = ":".join(f"{x:02X}" for x in mac_parts)
+
+        player_name = f"player-name{host.replace('.', '')}"
+
+        player = await AsyncMock(spec=Player)()
+        player.__aenter__.return_value = player
+
+        status_store = ValueStore(
```

Comment:
I am a little bit puzzled why this is needed

Reply:
The integration is long-polling. It awaits `player.status` for at most 120 seconds and returns on change. It also calls it as often as possible without ant timeout in between.

I had to create something that does not return instantly all the time, but returns instantly once i change the underlying value.

There might be a better solution to do this :shrug:

Reply:
Why do we need to update the values immediately? Can we patch the code on the fly?

Reply:
This is probably a common problem. I am trying to test a long polling API. I cannot patch the method(`status`) because it has already been called and the resulting `Awaitable` is already in use. Patching the method would not do anything.

There is one alternative. I could instantly return an value without waiting, but that would probably result in 100% CPU utilization. Because that method would be called without any pause in between calls.

Or maybe a sensible in between: Returning after a very short period.

Reply:
Why would it reach 100% CPU?

Reply:
The loop boils down to:
```python
while True:
    await player.status()
```

Normally `status()` blocks until something changes. When `status()` is mocked with an instantly returning a value, it would be called in a loop without any pause in between. Which would probably block the CPU.

Reply:
And in the rest of the platform we query `self.player.something`?

Reply:
There is an other loop for `sync_status`. The remaining part is simple calls to `self.player.something`.

Reply:
I have thought about it and i can not think of any other solution to mock those function. (I renamed `ValueStore` to `LongPollingMock`)